### PR TITLE
Fix autopipette offset taking apply on mode switch into account

### LINKED ISSUE #850: DAP_ChangeHeadstageMode: Allow to ignore "apply on mode switch" when called because of internal reasons

### LINKED ISSUE #860: Fix Auto pipette offset and friends
They need to change the clamp mode, but clamp mode change does currently take "Apply on mode switch" into account. This is wrong and should be fixed. Add tests as well.

### PR DESCRIPTION
**Pull request by t-b**
Close #860.
Close #850.

We still respect "Mode switch via I=0" when changing clamp mode for the auto pipette offset function. Is that correct?

### CODE REVIEW

File: Packages/MIES/MIES_Constants.ipf [hunk 1]
```diff
@@ -673,12 +673,14 @@ Constant AMPLIFIER_CONNECTION_MCC_FAILED = 2 ///< calling MCC_SelectMultiClamp70
...
-/// @{
-Constant DO_MCC_MIES_SYNCING   = 0x0
-Constant SKIP_MCC_MIES_SYNCING = 0x1
-Constant NO_SLIDER_MOVEMENT    = 0x2
+/// @name Constants for the options parameter of DAP_ChangeHeadStageMode()
+/// @anchor ClampModeChangeOptions
+/// @{
+Constant DO_MCC_MIES_SYNCING   = 0x0 ///< Default mode with all bells and whistles
+Constant NO_SLIDER_MOVEMENT    = 0x2 ///< Does not move the headstage slider
+Constant CMC_SKIP_UPDATES      = 0x4 ///< Skips all unnecessary updates. Intereseting for temporarily switching the clamp mode,
```
Comment (timjarsky) [inline Packages/MIES/MIES_Constants.ipf:681]: @t-b **MCC**_SKIP_UPDATES?

Comment (t-b) [reply Packages/MIES/MIES_Constants.ipf:681]: @timjarsky Yeah good point, the constants should follow the same  naming scheme.

Comment (t-b) [reply Packages/MIES/MIES_Constants.ipf]: Done.

**Review by timjarsky — APPROVED** [review]

**Comment by timjarsky** [comment]
@t-b after this is merged. Could we backport to the latest release?

**Comment by t-b** [comment]
@timjarsky See #866.

**Pull request merged**